### PR TITLE
release-2.1: assorted missing telemetry

### PR DESCRIPTION
**Pull request by knz**
Backport:
  * 4/4 commits from "sql/parser: add missing telemetry for unimplemented features" (#31635)
  * 1/1 commits from "pgwire: add telemetry for fetch limits" (#31637)
  * 1/1 commits from "sql,telemetry: report built-in function name in eval failures" (#31677)

Please see individual PRs for details.

/cc @cockroachdb/release

### CODE REVIEW

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -8355,6 +8376,7 @@ unreserved_keyword:
 | DAY
 | DEALLOCATE
 | DELETE
+| DEFERRED
```
Comment (bdarnell) [inline pkg/sql/parser/sql.y]: Grammar changes including new keywords after RC1? I think this missed the cutoff and needs to wait for 2.2. Telemetry work really needs to be planned as a part of the main development cycle; it's not something we can do post-beta when we're only supposed to be making stability-improving changes. @awoods187

**Comment by awoods187** [comment]
@bdarnell it wasn't planned this late in the cycle it was a minor oversight. It's really important for us to have good telemetry in the initial major release because many users don't upgrade to point releases--or at least not as frequently as we urge them to do so.

File: pkg/sql/parser/sql.y (hunk 1 above)
Comment (knz) [reply pkg/sql/parser/sql.y]: This is an unreserved keyword. It doesn't change the set of queries that will be accepted nor the set of queries that produce errors.

**Comment by knz** [comment]
@BramGruneir can you review this PR today? This may be good to get into 2.1.1.

**Review by BramGruneir — APPROVED** [review]
:lgtm:

As long as this has @bdarnell's blessing.

> Reviewed 2 of 2 files at r1, 3 of 3 files at r2, 2 of 2 files at r3, 3 of 3 files at r4, 4 of 4 files at r5, 4 of 4 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31638#-:-LR8JhiCAT-eUnvdniPA:bm64uu4)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by awoods187** [comment]
let's definitely get this into 2.1.1 cc @BramGruneir

**Comment by knz** [comment]
TC is wedged on a stress test somehow...

**Comment by knz** [comment]
ah, it is done.

**Pull request merged**